source/encoding.c

```diff
@@ -226,6 +268,13 @@ int aws_base64_encode(const struct aws_byte_buf *AWS_RESTRICT to_encode, struct
         return aws_raise_error(AWS_ERROR_SHORT_BUFFER);
     }
 
+    if (aws_common_private_has_b64_simd()) {
```

Comment:
Why is this a runtime rather than a compile-time check?

Reply:
Because we may be executing on a CPU with different capabilities than the one that built the code.

Reply:
But then you'll have to re-arrange your code, as both the run-time detection as the actual code using the extensions are compiled with the same options (they are in the same file). From https://gcc.gnu.org/onlinedocs/gcc-4.8.5/gcc/X86-Built-in-Functions.html: "If you specify command-line switches such as -msse, the compiler could use the extended instruction sets even if the built-ins are not used explicitly in the program. For this reason, applications that perform run-time CPU detection must compile separate files for each supported architecture, using the appropriate flags. In particular, the file containing the CPU detection code should be compiled without these options."

Reply:
Correct. Using the flags for encoding.c is a bug in CMakeLists. SIMD.cmake configures the flags for the file under source/arch/ differently.